CRM/Activityreport/Data.php

```diff
@@ -188,6 +177,27 @@ protected static function populateMultiValuesRow(array $row, array $fields, $off
   }
 
   /**
+   * Get number of multivalues combinations for given Activity row.
+   *
+   * @param array $row
+   *   Activity row
+   * @param array $fields
+   *   Array containing all Activity fields
+   * @return int
+   */
+  protected static function getTotalCombinations(array $row, array $fields) {
```

Comment:
I don't see a need for this method to be protected , private should be enough, same apply to the rest of the class properties and methods I don't see any need for them to be protected  for now and all should be private